### PR TITLE
magento/devdocs#: New topic. Product layouts

### PR DESCRIPTION
**Pull request by atwixfirster**
## Purpose of this pull request

This pull request (PR) adds new "Product layout" topic into front-end developer guide.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.1/frontend-dev-guide/layouts/product-layouts.html
- https://devdocs.magento.com/guides/v2.2/frontend-dev-guide/layouts/product-layouts.html
- https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/layouts/product-layouts.html

## Links to Magento source code
- [catalog_product_view.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Catalog/view/frontend/layout/catalog_product_view.xml)
- [catalog_product_view_type_bundle.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Bundle/view/frontend/layout/catalog_product_view_type_bundle.xml)
- [catalog_product_view_type_configurable.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/ConfigurableProduct/view/frontend/layout/catalog_product_view_type_configurable.xml)
- [catalog_product_view_type_downloadable.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Downloadable/view/frontend/layout/catalog_product_view_type_downloadable.xml)
- [catalog_product_view_type_grouped.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/GroupedProduct/view/frontend/layout/catalog_product_view_type_grouped.xml)
- [catalog_product_view_type_simple.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Catalog/view/frontend/layout/catalog_product_view_type_simple.xml)
- [catalog_product_view_type_virtual.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Catalog/view/frontend/layout/catalog_product_view_type_virtual.xml)
- [checkout_cart_configure.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Checkout/view/frontend/layout/checkout_cart_configure.xml)
- [checkout_cart_configure_type_bundle.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Bundle/view/frontend/layout/checkout_cart_configure_type_bundle.xml)
- [checkout_cart_configure_type_configurable.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/ConfigurableProduct/view/frontend/layout/checkout_cart_configure_type_configurable.xml)
- [checkout_cart_configure_type_downloadable.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Downloadable/view/frontend/layout/checkout_cart_configure_type_downloadable.xml)
- [checkout_cart_configure_type_simple.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Checkout/view/frontend/layout/checkout_cart_configure_type_simple.xml)
- [wishlist_index_configure.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Wishlist/view/frontend/layout/wishlist_index_configure.xml)
- [wishlist_index_configure_type_bundle.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Wishlist/view/frontend/layout/wishlist_index_configure_type_bundle.xml)
- [wishlist_index_configure_type_configurable.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Wishlist/view/frontend/layout/wishlist_index_configure_type_configurable.xml)
- [wishlist_index_configure_type_downloadable.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Wishlist/view/frontend/layout/wishlist_index_configure_type_downloadable.xml)
- [wishlist_index_configure_type_grouped.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Wishlist/view/frontend/layout/wishlist_index_configure_type_grouped.xml)
- [wishlist_index_configure_type_simple.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Wishlist/view/frontend/layout/wishlist_index_configure_type_simple.xml)

Thank you!

whatsnew 
Added a new topic for [Product layouts](https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/layouts/product-layouts.html) in the Frontend Developers Guide.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.1/frontend-dev-guide/layouts/product-layouts.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+---
+
+This topic provides information about product layouts files. Magento allows to customize view pages for all product types in the common layout files or particular product types.
+
+## Product view page 
+ 
+Layout file | Description
+--- | ---
+`catalog_product_view.xml` | Common layout. Affects all product types
+`catalog_product_view_type_bundle.xml` | Layout directives from this file implements to `bundle` product only
```
Comment (rogyar) [inline guides/v2.1/frontend-dev-guide/layouts/product-layouts.md]: Please, change the "directive" term to the "layout" or "node". The directive term in development has a little bit different meaning and more applicable to an argument or a property. 
So in result we have something like:

"Layout from this file is applied to bundle products only" and so on.

Comment (atwixfirster) [reply guides/v2.1/frontend-dev-guide/layouts/product-layouts.md]: Good point, @rogyar !

Thank you!

**fixed**

**Comment by atwixfirster** [comment]
> Please, provide also links to the existing layout files from Magento Github repository (for all examples) in a separate comment to this PR.
> 
> Thank you!

## Source links
- [catalog_product_view.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Catalog/view/frontend/layout/catalog_product_view.xml)
- [catalog_product_view_type_bundle.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Bundle/view/frontend/layout/catalog_product_view_type_bundle.xml)
- [catalog_product_view_type_configurable.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/ConfigurableProduct/view/frontend/layout/catalog_product_view_type_configurable.xml)
- [catalog_product_view_type_downloadable.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Downloadable/view/frontend/layout/catalog_product_view_type_downloadable.xml)
- [catalog_product_view_type_grouped.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/GroupedProduct/view/frontend/layout/catalog_product_view_type_grouped.xml)
- [catalog_product_view_type_simple.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Catalog/view/frontend/layout/catalog_product_view_type_simple.xml)
- [catalog_product_view_type_virtual.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Catalog/view/frontend/layout/catalog_product_view_type_virtual.xml)
- [checkout_cart_configure.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Checkout/view/frontend/layout/checkout_cart_configure.xml)
- [checkout_cart_configure_type_bundle.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Bundle/view/frontend/layout/checkout_cart_configure_type_bundle.xml)
- [checkout_cart_configure_type_configurable.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/ConfigurableProduct/view/frontend/layout/checkout_cart_configure_type_configurable.xml)
- [checkout_cart_configure_type_downloadable.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Downloadable/view/frontend/layout/checkout_cart_configure_type_downloadable.xml)
- [checkout_cart_configure_type_simple.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Checkout/view/frontend/layout/checkout_cart_configure_type_simple.xml)
- [wishlist_index_configure.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Wishlist/view/frontend/layout/wishlist_index_configure.xml)
- [wishlist_index_configure_type_bundle.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Wishlist/view/frontend/layout/wishlist_index_configure_type_bundle.xml)
- [wishlist_index_configure_type_configurable.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Wishlist/view/frontend/layout/wishlist_index_configure_type_configurable.xml)
- [wishlist_index_configure_type_downloadable.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Wishlist/view/frontend/layout/wishlist_index_configure_type_downloadable.xml)
- [wishlist_index_configure_type_grouped.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Wishlist/view/frontend/layout/wishlist_index_configure_type_grouped.xml)
- [wishlist_index_configure_type_simple.xml](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Wishlist/view/frontend/layout/wishlist_index_configure_type_simple.xml)

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by lorikrell** [comment]
running tests

**Comment by lorikrell** [comment]
running tests

**Pull request merged**